drivers/display/display_st7789h2.c

```diff
@@ -0,0 +1,647 @@
+			 const struct display_buffer_descriptor *desc,
+			 const void *buf)
+{
+	const struct st7789h2_config *config = dev->config;
+
+	__ASSERT(desc->width <= desc->pitch, "Pitch is smaller then width");
+	__ASSERT((desc->pitch * ST7789H2_PIXEL_SIZE * desc->height) <= desc->buf_size,
+			"Input buffer to small");
+
+	LOG_DBG("Writing %dx%d (w,h) @ %dx%d (x,y) p=%d n=%d",
+			desc->width, desc->height, x, y, desc->pitch, desc->buf_size);
+
+	st7789h2_set_mem_area(dev, x, y, desc->width, desc->height);
+
+	st7789h2_transmit(dev, ST7789V_CMD_RAMWR, NULL, 0);
+
+	st7789h2_set_cs(config, 1);
+	st7789h2_set_cmd(config, 0);
+
+	uint16_t *buffer = (void *)buf;
+	size_t len = desc->buf_size/ST7789H2_PIXEL_SIZE;
+
+	for (size_t i = 0; i < len; i++) {
+		st7789h2_data_bus_set(dev, buffer[i]);
+		st7789h2_wrn_pulse(config);
+	}
+
+	st7789h2_set_cs(config, 0);
+
+	return 0;
+}
+
+static void *st7789h2_get_framebuffer(const struct device *dev)
+{
+	return NULL;
+}
+
+static int st7789h2_set_brightness(const struct device *dev,
+			   const uint8_t brightness)
+{
+	return -ENOTSUP;
+}
+
+static int st7789h2_set_contrast(const struct device *dev,
+			 const uint8_t contrast)
+{
+	return -ENOTSUP;
+}
+
+static void st7789h2_get_capabilities(const struct device *dev,
+			      struct display_capabilities *capabilities)
+{
+	struct st7789h2_data *data = dev->data;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = data->width;
+	capabilities->y_resolution = data->height;
+
+	capabilities->supported_pixel_formats = PIXEL_FORMAT_RGB_565;
+	capabilities->current_pixel_format = PIXEL_FORMAT_RGB_565;
+
+	capabilities->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static int st7789h2_set_pixel_format(const struct device *dev,
+			     const enum display_pixel_format pixel_format)
+{
+
+	if (pixel_format == PIXEL_FORMAT_RGB_565) {
+		return 0;
+	}
+	LOG_WRN("Pixel format change not implemented");
+	return -ENOTSUP;
+}
+
+static int st7789h2_set_orientation(const struct device *dev,
+			    const enum display_orientation orientation)
+{
+	if (orientation == DISPLAY_ORIENTATION_NORMAL) {
+		return 0;
+	}
+	LOG_WRN("Changing display orientation not implemented");
+	return -ENOTSUP;
+}
+
+static void st7789h2_lcd_init(const struct device *dev)
+{
+	const struct st7789h2_config *config = dev->config;
+	struct st7789h2_data *data = dev->data;
+	uint8_t tmp;
+
+	st7789h2_set_lcd_margins(data, data->x_offset,
+				data->y_offset);
+
+	st7789h2_transmit(dev, ST7789V_CMD_CMD2EN, (uint8_t *) config->cmd2en_param,
+			 sizeof(config->cmd2en_param));
+
+	st7789h2_transmit(dev, ST7789V_CMD_PORCTRL, (uint8_t *) config->porch_param,
+			 sizeof(config->porch_param));
+
+	/* Digital Gamma Enable, default disabled */
+	tmp = 0x00;
+	st7789h2_transmit(dev, ST7789V_CMD_DGMEN, &tmp, 1);
+
+	/* Frame Rate Control in Normal Mode, default value */
+	tmp = 0x1e; /* 40 Hz */
+	st7789h2_transmit(dev, ST7789V_CMD_FRCTRL2, &tmp, 1);
```

Comment:
why can't these be configured via DT?

Reply:
Inspired from ST7789V2 which didn't make them settable